File: kotlinapp/src/main/java/com/github/mrvilkaman/ui/screens/testfrags/Frag4ScreenFragment.java
package com.github.mrvilkaman.ui.screens.testfrags;

import com.github.mrvilkaman.di.ActivityCoreComponent;
import com.github.mrvilkaman.presentationlayer.fragments.core.BaseFragment;

public class Frag4ScreenFragment extends FragBaseScreenFragment {


	public static Frag4ScreenFragment open() {
		return new Frag4ScreenFragment();
	}

	@Override
	protected int getNumber() {
		return 4;
	}

	@Override
	protected BaseFragment nextFragment() {
		return Frag5ScreenFragment.Companion.open();
	}

	@Override
	protected int getIcon() {
		return android.R.drawable.ic_dialog_map;
	}

	@Override
	public void daggerInject() {
		ActivityCoreComponent component = getComponent(ActivityCoreComponent.class);
		DaggerFragScreenComponent.builder().activityCoreComponent(component).build().inject(this);
	}
}

File: kotlinapp/src/main/java/com/github/mrvilkaman/ui/screens/testfrags/Frag4ScreenFragment.kt
package com.github.mrvilkaman.ui.screens.testfrags

import com.github.mrvilkaman.ui.screens.ScreenKey

class Frag4ScreenFragment : FragBaseScreenFragment() {

    override val number: Int
        get() = 4

    override val icon: Int
        get() = android.R.drawable.ic_dialog_map

    override fun nextScreenKey(): String? = ScreenKey.FRAG5


    companion object {


        fun open(): Frag4ScreenFragment {
            return Frag4ScreenFragment()
        }
    }
}
